chapter2/integration_with_MIQ.md

```diff
@@ -1,4 +1,92 @@
+
+In Default tab, We need to configure the host and authenticate  details of OpenStack provider.
+
+- To authenticate the provider, Select a security protocol method. SSL allows you to authenticate the provider securely using a certificate authority.
+
+- Enter the hostname or IP Address either IPv4 or IPv6 of the provider.
+
+- In API port, OpenStack uses 5000 port by default.
+
+- Fill in the credentials in Username and password field. The usrename must be privileged and its corresponding password.
+
+| Note |You will get the password in keystonerc_admin file in /root directory.|
+|------|:------|
+
+- Validate the details to make sure that ManageIQ can connect to OpenStack provider.
+
+![Fig 2-Add OpenStack Provider-Default_Tab](../images/chapter2/Add_OpenStack_Provider-Default_Tab.png "Add OpenStack Provider - Default Tab")
+
+In Events tab under Endpoints section, We need to configure how ManageIQ should receive events from the OpenStack provider.
+
+- If you prefer to use AMQP messaging bus, Select AMQP event service. For this you need to set the public port used by AMQP in API Port and the corresponding credentials. Also validate the credentials to confirm them.
+
+- If you want to use Telemetry service by the OpenStack provider, Select Ceilometer event service.
+
+| Note |To use Ceilometer service you must configure the *notification_driver*. To do so edit the *undercloud.conf* and set `store_events = true` |
+|------|:------|
+
+![Fig 3-Add OpenStack Provider-Event Tab](../images/chapter2/Add_OpenStack_Provider-Event_Tab.png "Add OpenStack Provider - Event Tab")
+
+Here your cloud provider is successfully integrated with ManageIQ. All the instances, images, networks, security groups, etc. are summarized here. It also displays the machine status with its last session details.
+
+| Note |It takes time to integrate the cloud provider and display all its details.|
+|------|:------|
+
+![Fig 4-OpenStack Summary](../images/chapter2/OpenStack_Summary.png "OpenStack Summary")
+
+The web interface uses virtual thumbnails to represent infrastructure providers. Each thumbnail contains four quadrants by default, which display basic information about each provider:
+
+1 Number of hosts<br>
+2 Management system software<br>
+3 Currently used<br>
+4 Authentication status
+
+![Fig 5-OpenStack Thumbnail](../images/chapter2/OpenStack_Thumbnail.png "OpenStack Thumbnail")
+
+You can walk through all these things by just clicking on it.
+
+![Fig 6-OpenStack Instances](../images/chapter2/OpenStack_Instances.png "OpenStack Instances")
+
+<br>
+
+---
+
+### Issue Faced
+
+We faced a issue after the integration of the OpenStack provider. Even after the integration, the details and information of OpenStack provider like instances, images, network, etc were not displayed.
+
+But later we discovered that there were compatibility issues of *Fog Library* in the OpenStack provider with ManageIQ and fog is the plugin of Ruby Gem to talk to OpenStack clouds.
```

Comment:
Instead of 'fog is the plugin of Ruby Gem', 'fog is the Ruby cloud services library' is correct statement.